test/src/crypto.js

```diff
@@ -153,4 +153,136 @@ describe('crypto', () => {
       expect(await crypto.decrypt(encrypted)).to.be('I am a string');
     });
   });
+
+  describe('Using AAD', () => {
+    describe('type validation', () => {
+      describe('for encrypt', () => {
+        it('should throw an error if aad is empty string', async () => {
```

Comment:
`expect.js` has `throw`/`throwException`/`throwError`, can we not use those here, and elsewhere in these tests where decryption/encryption fails?

Reply:
Ah so it does, yeah I can refactor the tests to use those!

Reply:
I cannot seem to get expectjs to work with the promises. I had looked for a `rejected` option which it does not seem to support. Any suggestions because it would be nicer than the try...catch blocks

Reply:
> I cannot seem to get expectjs to work with the promises

😞 that's a bummer. We might be stuck doing our own solution, or migrating all tests to use `jest` or another maintained "assertion library". 

For the time being, how would you feel about introducing a `expectRejected` so we at least don't have to replicate this logic multiple times throughout this file? We could also potentially use the `finally` clause to make this a bit friendlier (this is a suggestion/nit, feel free to ignore it):


```
const expectRejected = async (promise, expectedError) => {
  let error = null;
  try {
    await promise;
  }  catch (err) {
    err = error;
  } finally {
    expect(er).not.to.be(null);
    expect(er).to.match(expectedError);
  }
}
```

Reply:
It'd be nice to switch to `jest` but I don't think it should be done in this PR. I don't particularly like adding custom expects functions unless they make the tests compatible with the future proposed syntax.

Reply:
> I don't particularly like adding custom expects functions unless they make the tests compatible with the future proposed syntax.

Do you have an approach in mind that would do so?

Reply:
I attempted to extend `expect.js` that did not go so well. I also took the time to update the package to use jest see #5 and can update these tests based on that PR and this one. I think that is the best direction to go in.